### ISSUE TITLE
Unknown `keys` in `Forc.toml` print the same warning for each compiled library

### ISSUE THREAD

**Issue by eureka-cpu** [FuelLabs/sway#928]
Found while testing #918 

>adding an unused key [[foo]], as intended gets: `WARNING! unused manifest key: foo`, however I get that warning twice. Once at the beginning of the build, and once at the end before `Compiled script "address_test"`. I think what's happening is that it's passing the warning for each library that gets compiled...

Edit: The issue may be that we're reading the `manifest` a second time in `forc_build::BuildPlan` which is causing the warning to appear twice.

**Comment by eureka-cpu** [comment]
>Ah this is likely due to Manifest::validate method getting called within the Manifest constructor, and the constructor getting called multiple times during a forc build. Ideally the Manifest should only be constructed once per package, but its convenient for some of the graph related code to construct it on demand. This would be easier to address with https://github.com/FuelLabs/sway/issues/885.

**Comment by eureka-cpu** [comment]
#4709 is actually a duplicate of this issue and @gr00vytvniks is working on it I believe

**Comment by kayagokalp** [comment]
#4709 and this one seems unrelated and i am not really sure if this is a problem. If the same issue is present in each compiler library shouldn't we print that issue for each of them. 

If you believe the same warning or error for the same package gets printed twice, then that is a problem but I believe `from_file` constructors (which prints these warnings) are called only once and then we are using a in-memory cache for reaching manifests (`ManifestMap`) basically it is done like that so that we do not do extensive I/O. If i am not mistaken this should not happen after #3368

**Comment by eureka-cpu** [comment]
Figured this out with @kayagokalp it's no longer an issue
